src/runtime/HalidePyTorchCudaHelpers.h

```diff
@@ -0,0 +1,68 @@
+#ifndef HL_PYTORCH_CUDA_HELPERS_H
+#define HL_PYTORCH_CUDA_HELPERS_H
+
+/** \file
+ * Override Halide's CUDA hooks so that the Halide code called from PyTorch uses 
+ * the correct GPU device and stream.
+ */
+
+#ifdef HL_PT_CUDA
+#include "HalideRuntimeCuda.h"
+#include "cuda.h"
+
+namespace Halide {
+namespace PyTorch {
+
+typedef struct UserContext {
+  UserContext(int id, CUcontext *ctx, cudaStream_t* stream) :
```

Comment:
four-space indents for all C++ code in Halide, please

Reply:
ok

Reply:
ok